File: CSE 241 (OOP - C++ - Java)/Hw5/CSE241_mehmetHuseyin_yildiz_200104004095/Location.cpp
#include "Location.h"

Location::Location()
{
    setLocation(-1, -1);
}

Location::Location(int x, int y)
{
    setLocation(x, y);
}

void Location::set_x(int x_)
{
    x = x_;
}

void Location::set_y(int y_)
{
    y = y_;
}

int Location::get_y() const
{
    return y;
}

int Location::get_x() const
{
    return x;
}

void Location::setByAdding(int x_, int y_)
{
    x += x_;
    y += y_;
}

Location Location::changeLocation(int x_, int y_)
{
    Location newLocation;
    newLocation.set_x(x + x_);
    newLocation.set_y(y + y_);
    return newLocation;
}

bool Location::operator==(Location another) const
{
    return (x == another.get_x() && y == another.get_y());
}

std::istream& operator>>(std::istream& in, Location location)
{
    int a, b;
    in >> a;
    in >> b;
    location.set_x(a);
    location.set_y(b);
    return in;
}

File: CSE 241 (OOP - C++ - Java)/Hw7/Cse241_Hw7/Location.java
package Cse241_Hw7;

import java.io.Serializable;

public class Location implements Serializable,Cloneable {
	

	/**
	 * 
	 */
	private static final long serialVersionUID = 7556794030623375123L;
	private int x;
	private int y;
	
	
	public Location() {	    setLocation(-1, -1);	}
	
	public Location(int x, int y) {     setLocation(x, y); }
	
	public void setLocation(int x_, int y_) { x = x_; y = y_; }
	
	public void set_x(int x_) { x = x_; }
	
	public void set_y(int y_) { y = y_; }
	
	public int get_y() { return y; }
	
	public int get_x() { return x; }
	
	public void setByAdding(int x_, int y_)  
	{
		x += x_;
		y += y_; 
	}
	
	public Location changeLocation(int x_, int y_)  
	{
	    Location newLocation = new Location();
	    newLocation.set_x(x + x_);
	    newLocation.set_y(y + y_);
	    return newLocation;
	}
	
	// public friend std::istream& operator>> (std::istream& in, Location location);

	// bool operator==(Location another)const;
	
	
	
	@Override
	protected Object clone() throws CloneNotSupportedException {
		return super.clone();
	}


}


